dace/properties.py

```diff
@@ -554,6 +554,39 @@ def from_json(self, data, sdfg=None):
         return list(map(self.element_type, data))
 
 
+class TransformationHistProperty(Property):
+    """ Property type for transformation histories.
+    """
+    def __init__(self, *args, **kwargs):
+        """
+        Create a List property with element type Transformation.
+        :param args: Other arguments (inherited from Property).
+        :param kwargs: Other keyword arguments (inherited from Property).
+        """
+
+        kwargs['dtype'] = list
+        super().__init__(*args, **kwargs)
+
+    def __set__(self, obj, val):
+        super(TransformationHistProperty, self).__set__(obj, val)
+
+    def to_json(self, hist):
+        if hist is None:
+            return None
+        return [elem.to_json() for elem in hist]
+
+    def from_json(self, data, sdfg=None):
+        if data is None:
+            return data
+        if not isinstance(data, list):
+            raise TypeError(
+                'TransformationHistProperty expects a list input, got %s' % data
+            )
+        # This is imported here to avoid an import loop.
+        from dace.transformation.pattern_matching import Transformation
+        return [Transformation.from_json(elem) for elem in data]
```

Comment:
I think you should use `dace.serialize.from_json` because it can detect Transformations, expansions, and SubgraphTransformations